public/app/core/services/context_srv.ts

```diff
@@ -157,9 +157,9 @@ export class ContextSrv {
 
   hasAccessToExplore() {
     if (this.accessControlEnabled()) {
-      return this.hasPermission(AccessControlAction.DataSourcesExplore);
+      return this.hasPermission(AccessControlAction.DataSourcesExplore) && config.exploreEnabled;
     }
-    return (this.isEditor || config.viewersCanEdit) && config.exploreEnabled;
+    return (this.isEditor || config.viewersCanEdit) && !(config.exploreEnabled === false);
```

Comment:
```suggestion
    return (this.isEditor || config.viewersCanEdit) && (config.exploreEnabled !== false);
```

Reply:
Youre absolutely right, we should just stick with the existing simple logic check. I changed it back to what it was.